### PR TITLE
Handling when `TestFile` is called on a Go test in a subdirectory

### PR DESCRIPTION
**Pull request by codeinabox**

### CODE REVIEW

**Comment by codeinabox** [comment]
@janko-m I've written a scenario to cover the scenario that is causing my tests in a project not to run but what is the best way to implement this?

**Comment by janko-m** [comment]
I'm not that familiar with Go, so I'm not sure. Did I get it right that this test is currently passing, but a test file on your project is still not being run?

**Comment by codeinabox** [comment]
Currently it doesn't support running tests in a subdirectory when doing `TestFile` as `go test` only runs tests in the current directory. I've added a new test to indicate how it should behave but not sure how to implement it as my Vimscript is not so good.

**Comment by janko-m** [comment]
Aha, now I get it. You probably want `fnamemodify()`, it has a lot of options, so you should probable be able find what you need there.

**Comment by codeinabox** [comment]
Done. This passes the tests though maybe can be written more elegantly

File: autoload/test/go/gotest.vim [hunk 1]
```diff
@@ -12,7 +12,8 @@ function! test#go#gotest#build_position(type, position) abort
     if !empty(name) | let name = '-run '.shellescape(name, 1) | endif
     return [name]
   elseif a:type == 'file'
-    return []
+    let path = fnamemodify(a:position['file'], ':h.')
+    return path != '.' ? ['./' . path . '/...'] : []
```
Comment (janko-m) [inline autoload/test/go/gotest.vim]: Hmm, what does the dot mean? Is it maybe better to write it like this?

```vim
let path = fnamemodify(a:position['file'], ':h')
return !empty(path) ? ['./' . path . '/...'] : []
```

Comment (codeinabox) [inline autoload/test/go/gotest.vim]: I don't need the dot in `fnamemodify` though I do need to check if the path is equal to "." which is the current directory. Have updated my PR accordingly.

**Comment by janko-m** [comment]
Great, thanks for the patch!

**Pull request merged**